Client/Frontend/Home/TopSites/DataManagement/TopSitesManager.swift

```diff
@@ -116,6 +128,12 @@ class TopSitesManager: FeatureFlaggable, HasNimbusSponsoredTiles {
         refreshIfNeeded(forceTopSites: false)
     }
 
+    private func getAvailableSpaceCount(numberOfTilesPerRow: Int) -> Int {
+        let pinnedSiteCount = countPinnedSites(sites: historySites)
+        let totalNumberOfShownTiles = numberOfTilesPerRow * numberOfRows
+        return totalNumberOfShownTiles - pinnedSiteCount
+    }
```

Comment:
nit: It would be good to write a comment to explain how this counting logic works

q. Do we need to do any specific counts for iPad vs iPhone? (just curious)

Reply:
Yeah we do, the specific counting for iPad versus iPhone happens with numberOfTilesPerRow which is passed down from the TopSitesViewModel. It's documented a bit in that viewModel. Let me know if you think we should supplement the information there